wade-review.md

```diff
@@ -0,0 +1,596 @@
+                            stateCounter = new GitHubPRStateCounter();
+                        }
+                        if (prInfo.state().equalsIgnoreCase("open")) {
+                            stateCounter.setOpen(stateCounter.getOpen() + 1);
+                        } else {
+                            stateCounter.setClosed(stateCounter.getClosed() + 1);
+                        }
+                        kvStore.put(STORE_KEY, stateCounter);
+                    }
+```
+
+Then, we update the store.
+
+```java
+
+    static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> {
+
+        static final String STORE_KEY = "state-counter";
+
+        @Override
+        public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+            return new Processor<>() {
+                private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+                }
+
+                @Override
+                public void process(final Record<String, GitHubPRInfo> record) {
+                    GitHubPRInfo prInfo = record.value();
+                    if(!prInfo.state().equals(NO_STATE)) {
+                        GitHubPRStateCounter stateCounter = kvStore.get(STORE_KEY);
+                        if (stateCounter == null) {
+                            stateCounter = new GitHubPRStateCounter();
+                        }
+                        if (prInfo.state().equalsIgnoreCase("open")) {
+                            stateCounter.setOpen(stateCounter.getOpen() + 1);
+                        } else {
+                            stateCounter.setClosed(stateCounter.getClosed() + 1);
+                        }
+                        kvStore.put(STORE_KEY, stateCounter);
+                    }
+                }
+            };
+        }
+
+        @Override
+        public Set<StoreBuilder<?>> stores() {
+            return Collections.singleton(Stores.keyValueStoreBuilder(Stores.inMemoryKeyValueStore(STORE_NAME), Serdes.String(), prStateCounterSerde));
+        }
+    }
+
+}
+```
+
+Lastly, we use the processor API to create our state store. We return a singleton with the store values built from our store builder. 
+
+
+## The GitHub PR Ratio class
+
+```java
+public class GitHubPrRatio {
+}
+```
+
+ Next, I’m going to create a GitHubPRRatio class to hold this processing… process!. I’ll need some variables:
```

Comment:
While I appreciate the wordplay, I don't really understand at this point what the job of the GitHubPrRatio class is.

Is it's job just to hold the processor? In that case I'd name it "ProcessorHolder" or something. I suspect there is more going on and I'd be looking for an explanation of that.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/548fd1a961a35a7758d2588e269fc26e3d1e07fe